src/component/legend/LegendView.js

```diff
@@ -361,9 +452,52 @@ export default echarts.extendComponentView({
         );
 
         var contentRect = contentGroup.getBoundingRect();
-        contentGroup.attr('position', [-contentRect.x, -contentRect.y]);
+        var contentPos = [-contentRect.x, -contentRect.y];
+
+        if (selectorGroup) {
+            // Place buttons in selectorGroup
+            selectorGroup && layoutUtil.box(
+                // Buttons in selectorGroup always layout horizontally
+                'horizontal',
+                selectorGroup,
+                legendModel.get('selectorItemGap', true)
+            );
+
+            var selectorRect = selectorGroup.getBoundingRect();
+            var selectorPos = [-selectorRect.x, -selectorRect.y];
+
+            var selectorLegendGap = retrieve2(
+                legendModel.get('selectorLegendGap', true), legendModel.get('itemGap', true)
+            );
+
+            var orientIdx = legendModel.getOrient().index;
+            var WH = orientIdx === 0 ? 'width' : 'height';
```

Comment:
The original way is:

```js
var WH = ['width', 'height']; // as a constant
var XY = ['x', 'y']; // as a constant

function someFn() {
    var orientIdx = legnedModel.getOrient().index;
    var wh = WH[orientIdx];
    var hw = WH[1 - orientIdx]
    var xy = XY[orientIdx];

    ...
}
```

Reply:
I know the original, but I think that in the same legend component, as long as their logic is the same, it doesn't have to be written same.